build_runner/lib/src/generate/build_impl.dart

```diff
@@ -222,12 +226,13 @@ class BuildImpl {
       // Skip the build step if none of the outputs need updating.
       var skipBuild = !builderOutputs.any((output) =>
           (_assetGraph.get(output) as GeneratedAssetNode).needsUpdate);
-      if (skipBuild) continue;
+      if (skipBuild) return;
       var wrappedReader = new SinglePhaseReader(
           _reader, _assetGraph, phaseNumber, _packageGraph.root.name);
       var wrappedWriter = new AssetWriterSpy(_writer);
       await runBuilder(
-          builder, [input], wrappedReader, wrappedWriter, _resolvers);
+          builder, [input], wrappedReader, wrappedWriter, _resolvers,
```

Comment:
`runBuilder` can handle multiple inputs - is there any way to structure this to use that?

Reply:
I can try that out sure, will follow up